src/metabase/models/data_permissions.clj

```diff
@@ -194,6 +203,14 @@
       (or (coalesce perm-type perm-values)
           (least-permissive-value perm-type)))))
 
+(mu/defn user-has-permission-for-database? :- :boolean
+  "Returns a Boolean indicating whether the user has the specified permission value for the given database ID and table ID,
+   or a more permissive value."
+  [user-id perm-type perm-value database-id]
+  (at-least-as-permissive? perm-type
```

Comment:
I thought you mentioned that if `perm-a` is more permissive than `perm-b`, `perm-a` does not necessarily grant all of the abilities granted by `perm-b` - did you reconsider that?

Reply:
(addressed on Slack)